### DISCUSSION TITLE
Problem with IStreamReader

### DISCUSSION THREAD

**Discussion by MartinZhechev** [LuckyPennySoftware/MediatR#798] [Q&A]
Hello, I'm really sorry if this is not the place for questions about a problem. I've been battling this problem for a few days and I can't find a solution anywhere.

I'm developing a .net 6 solution for my employer, which uses MediatR for CQRS and I'm really liking it, so first off, thanks for the effort!

For a year everything worked perfectly fine because I only used IRequest. But I now have a new requirement which in order to work the request must use IStreamRequest. But for some reason I get the exception:

**"Handler was not found for request of type MediatR.IStreamRequestHandler"**

I've double and quadruple checked everything. Every other request works perfectly fine, but when I call the StreamRequest one it fails... 

  ```
public class NotificationsStreamRequest : IStreamRequest<int>
  {
      public class NotificationsStreamRequestHandler : 
          IStreamRequestHandler<NotificationsStreamRequest, int>
      {
          public async IAsyncEnumerable<int> Handle(
              NotificationsStreamRequest request,
              [EnumeratorCancellation] CancellationToken cancellationToken)
          {
              while (!cancellationToken.IsCancellationRequested)
              {
                  await Task.Delay(60000, cancellationToken);
                  yield return 0;
              }
          }
      }
  }
```
And I've registered MediatR as such:

`services.AddMediatR(Assembly.GetExecutingAssembly())`

This is the test request I have, which throws the error. The IRequest requests have the exact same structure. Thanks in advance, have a great day! :)

**Comment by jbogard** [comment]
Probably because it's a nested class there? This isn't the right repo for this question, MediatR itself does not handle registration. That's the other library (MediatR.Extensions.DependencyInjection).
